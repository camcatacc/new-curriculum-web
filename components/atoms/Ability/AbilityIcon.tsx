// Modules
import React from "react";

// Elements
import CircularBackground from "components/HOC/withCircularBackground";
import { SvgIcon } from "@material-ui/core";

// Styles
import { primary, secondary } from "styles/customColors";

// Definitions
export interface AbilityIconProps {
	icon: string;
}

// Element
const AbilityIcon = ({ icon }: AbilityIconProps) => (
	<CircularBackground height="100%" width="100%" backgroundColor={secondary} padding="20%">
		<SvgIcon style={{ color: primary, height: "100%", width: "100%" }}>
			<path fill="currentColor" d={icon} />
		</SvgIcon>
	</CircularBackground>
);

export default AbilityIcon;
