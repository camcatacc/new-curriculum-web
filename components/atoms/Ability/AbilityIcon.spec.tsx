// Modules
import { render } from "@testing-library/react";
import React from "react";

// Elements
import { InsertChartOutlinedTwoTone } from "@material-ui/icons";
import AbilityIcon from "components/atoms/Ability/AbilityIcon";

// Tests
describe("AbilityIcon", () => {
	it("Renders as expected", () => {
		const { container } = render(<AbilityIcon icon={InsertChartOutlinedTwoTone} />);
		expect(container).toMatchSnapshot();
	});
});
