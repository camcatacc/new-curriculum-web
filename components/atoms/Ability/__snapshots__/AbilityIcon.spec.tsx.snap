// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbilityIcon Renders as expected 1`] = `
<div>
  <div
    class="rounded-full w-min h-min"
    style="background-color: rgb(0, 2, 67); height: 100%; width: 100%; padding: 20%;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      style="color: rgb(60, 235, 147); height: 100%; width: 100%;"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4zm2 2H5V5h14v14zm0-16H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
      />
    </svg>
  </div>
</div>
`;
